lxd/storage/backend_lxd.go

```diff
@@ -0,0 +1,351 @@
+package storage
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/lxc/lxd/lxd/migration"
+	"github.com/lxc/lxd/lxd/operations"
+	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/lxd/storage/drivers"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+)
+
+type lxdBackend struct {
+	driver drivers.Driver
+	id     int64
+	name   string
+	state  *state.State
+}
+
+func (b *lxdBackend) DaemonState() *state.State {
+	return b.state
+}
+
+func (b *lxdBackend) ID() int64 {
+	return b.id
+}
+
+func (b *lxdBackend) Name() string {
+	return b.name
+}
+
+func (b *lxdBackend) Driver() drivers.Driver {
+	return b.driver
+}
+
+func (b *lxdBackend) create(dbPool *api.StoragePool, op *operations.Operation) error {
+	revertPath := true
+
+	// Create the storage path.
+	path := drivers.GetPoolMountPoint(b.name)
+	err := os.MkdirAll(path, 0711)
+	if err != nil && !os.IsExist(err) {
+		return err
+	}
+	defer func() {
+		if !revertPath {
+			return
+		}
+
+		os.RemoveAll(path)
+	}()
+
+	// Create the storage pool on the storage device.
+	err = b.driver.Create()
+	if err != nil {
+		return err
+	}
+
+	// Mount the storage pool.
+	ourMount, err := b.driver.Mount()
+	if err != nil {
+		return err
+	}
+
+	// tomp TODO is this needed?
+	if ourMount {
+		defer b.driver.Unmount()
```

Comment:
@stgraber if the expectation is that all storage pools will be mounted at start or create time, then is this part still needed?

Reply:
Yes, the expectation is that both `GetByName` and `Create` will return you a `Pool` but neither will mount it for you. LXD on startup will call `GetByName` on all pools, followed by a `Mount()` call of each of them. Similarly, when calling `Create()`, you should follow that with `Mount()` if you intend to use it.